### PR TITLE
Add Tooling API model for declarative schemas

### PR DESCRIPTION
**Pull request by jbartok**
### Contributor Checklist
- [ ] [Review Contribution Guidelines](https://github.com/gradle/gradle/blob/master/CONTRIBUTING.md).
- [ ] Make sure that all commits are [signed off](https://git-scm.com/docs/git-commit#Documentation/git-commit.txt---signoff) to indicate that you agree to the terms of [Developer Certificate of Origin](https://developercertificate.org/).
- [ ] Make sure all contributed code can be distributed under the terms of the [Apache License 2.0](https://github.com/gradle/gradle/blob/master/LICENSE), e.g. the code was written by yourself or the original code is licensed under [a license compatible to Apache License 2.0](https://apache.org/legal/resolved.html).
- [ ] Check ["Allow edit from maintainers" option](https://help.github.com/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork/) in pull request so that additional changes can be pushed by Gradle team.
- [ ] Provide integration tests (under `<subproject>/src/integTest`) to verify changes from a user perspective.
- [ ] Provide unit tests (under `<subproject>/src/test`) to verify logic.
- [ ] Update User Guide, DSL Reference, and Javadoc for public-facing changes.
- [ ] Ensure that tests pass sanity check: `./gradlew sanityCheck`.
- [ ] Ensure that tests pass locally: `./gradlew <changed-subproject>:quickTest`.

### Reviewing cheatsheet

Before merging the PR, comments starting with 
- ❌ ❓**must** be fixed
- 🤔 💅 **should** be fixed
- 💭 **may** be fixed
- 🎉 celebrate happy things

### CODE REVIEW

**Comment by jbartok** [comment]
@bot-gradle test this

**Comment by jbartok** [comment]
@bot-gradle test this

File: platforms/core-configuration/declarative-dsl-tooling-models/src/main/java/org/gradle/declarative/dsl/tooling/models/DeclarativeSchemaModel.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+ * limitations under the License.
+ */
+
+package org.gradle.declarative.dsl.tooling.models;
+
+import org.gradle.declarative.dsl.schema.AnalysisSchema;
+
+public interface DeclarativeSchemaModel {
+
+    AnalysisSchema schema();
```
Comment (eskatos) [inline platforms/core-configuration/declarative-dsl-tooling-models/src/main/java/org/gradle/declarative/dsl/tooling/models/DeclarativeSchemaModel.java:23]: ❌ This is the schema for projects, the getter should reflect that fact given we will eventually add more for settings. Something like `getProjectsSchema()` would do.

**Comment by jbartok** [comment]
@bot-gradle test this

**Comment by jbartok** [comment]
@bot-gradle test this

**Comment by jbartok** [comment]
@bot-gradle test this

**Comment by jbartok** [comment]
@bot-gradle test this

**Comment by jbartok** [comment]
@bot-gradle test this

**Comment by jbartok** [comment]
@bot-gradle test this

File: platforms/core-configuration/declarative-dsl-provider/src/main/kotlin/org/gradle/internal/declarativedsl/provider/DeclarativeDslServiceRegistry.kt [hunk 2]
```diff
@@ -34,9 +36,16 @@ class DeclarativeDslServiceRegistry : AbstractPluginServiceRegistry() {
 
 internal
 object BuildServices {
+
+    private
+    val schemaRegistry = DefaultDeclarativeSchemaRegistry()
```
Comment (eskatos) [inline platforms/core-configuration/declarative-dsl-provider/src/main/kotlin/org/gradle/internal/declarativedsl/provider/DeclarativeDslServiceRegistry.kt:41]: ❌ We should not create this instance here. Instead the `createSchemaRegistry()` function below should create the instance and the `createDeclarativeKotlinScriptEvaluator()` function should take the registry as an injected parameter.

**Review by eskatos — CHANGES_REQUESTED** [review]
Looking good. Required one change in how we register a service.

File: platforms/core-configuration/declarative-dsl-provider/src/main/kotlin/org/gradle/internal/declarativedsl/provider/DeclarativeDslServiceRegistry.kt (hunk 2 above)
Comment (jbartok) [reply platforms/core-configuration/declarative-dsl-provider/src/main/kotlin/org/gradle/internal/declarativedsl/provider/DeclarativeDslServiceRegistry.kt]: Fixed, please approve.

**Review by eskatos — APPROVED** [review]

**Comment by jbartok** [comment]
@bot-gradle test RFM

**Pull request merged**